torch/utils/data/dataset.py

```diff
@@ -350,5 +351,5 @@ def random_split(dataset: Dataset[T], lengths: Sequence[int],
     if sum(lengths) != len(dataset):  # type: ignore
         raise ValueError("Sum of input lengths does not equal the length of the input dataset!")
 
-    indices = randperm(sum(lengths), generator=generator).tolist()
+    indices = index_utils.Permutation(len(dataset), generator.seed())
```

Comment:
Will using `generator.seed()` have the same behaviour as existing (i.e. passing the whole generator).

From the docs, it seems that `generator.seed()` actually generates a new seed and sets the generator with that seed.
https://pytorch.org/docs/stable/generated/torch.Generator.html#torch.Generator.seed

Reply:
The results are definitely going to change no matter what because I'm not pulling N numbers from the generator. Re-seeding it seems harmless as far as preserving determinism.

Not that it really matters, but I couldn't find another way to get a random number. Python's random cannot be imported because of package conflicts (not sure if I'm missing something), and generator just has the seed() method which mutates it.

Reply:
I think for random_split, as long as the result can preserve determinism as @almson said, it should be fine.
cc @VitalyFedyunin @SsnL @ejguan

Reply:
@kshitij12345 That was indeed the wrong use of generator.seed(). I've rewritten that part. It's too bad generator doesn't offer convenience methods to get a python scalar (or that there isn't a RNG for ordinary python code).